magento/inventory
# Fixes issue #2218 Backorder was not working on MSI

Fixed for #2218 issue: Because of inventory_sales module (MSI), Backorder Item Status is not showing if we placed order with backorder item in magento2.3.x 




### Description (*)

1. We found that method **checkQuoteItemQty** is reponsible for setting **qty_backordered** filed in db which displays **backordered** as "Item status".
2. By debugging we found that the original method **checkQuoteItemQty** was not calling. So after this found that this methous is intercepting with **aroundCheckQuoteItemQty**, by backtracing found that only this method was calling not going inside **checkQuoteItemQty** as  
`$result = $proceed($productId, $itemQty, $qtyToCheck, $origQty, $scopeId = null);` 
was missing.

### Manual testing scenarios (*)

1. Set a product up as backorderable - 0 quantity stock, "allow orders below 0", in stock, enabled
2. As a frontend customer, add this backordered product to a cart
3. Add an additional "normal" stock item to the cart (not backordered - this step may not be strictly required but is useful to make it easier to compare normal vs backordered line items)
4. Go through a normal customer frontend order process with this product
5. **Actual:** After placing the order, note that in the order history / admin order view that the order items list shows **ordered** in the "item status" column for **backorder** item and a **normal** item as well - **this is incorrect**
6. **Expected:** After placing the order, note that in the order history / admin order view that the order items list shows **ordered** in the "item status" column for a **normal** item and **backordered** in the "item status" column for a **backorder** item - **this is correct**

### Contribution checklist (*)
